### ISSUE TITLE
7.0: Warning: tsconfig does not affect esbuild's own target setting

### ISSUE THREAD

**Issue by RobinTail** [egoist/tsup#933]
How to get rid of this warning?

```
$ tsup
Warning: G] "tsconfig.json" does not affect esbuild's own target setting [tsconfig.json]

    node_modules/@tsconfig/node16/tsconfig.json:8:4:
      8 │     "target": "es2021",
        ╵     ~~~~~~~~

  This is because esbuild supports reading from multiple "tsconfig.json" files within a single build, and using different language targets for different files in the same build wouldn't be correct. If you want to set esbuild's language target, you should use esbuild's own global "target" setting such as with "target: 'es2021'".

CLI Building entry: src/index.ts
CLI Using tsconfig: tsconfig.json
CLI tsup v7.0.0
```

### Version

`7.0.0`

#### `tsconfig.json`

```json
{
  "extends": "@tsconfig/node16/tsconfig.json",
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "resolveJsonModule": true
  },
  "ts-node": {
    "swc": true
  }
}
```

#### `tsup.config.json`

```json
{
  "entry": ["src/index.ts"],
  "format": ["cjs", "esm"],
  "legacyOutput": true,
  "splitting": false,
  "sourcemap": true,
  "clean": true,
  "dts": true,
  "minify": true,
}
```

**Comment by prisis** [comment]
https://github.com/egoist/tsup/issues/930 would fix it, but currently esbuild is ignoring the field (for me)

**Comment by RobinTail** [comment]
The issue somehow was fixed by upgrading `esbuild` from v0.18.6 to v0.18.10.
I ran `yarn upgrade` command and there is no such warning anymore.
Perhaps it will help someone.
CC @prisis

**Comment by RobinTail** [comment]
I think it can be closed now, it seems that it was an `esbuild`'s issue.
